src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -138,6 +138,6 @@ private function isInlineableDefinition($id, Definition $definition, ServiceRefe
             return false;
         }
 
-        return true;
+        return $this->container->getDefinition($ids[0])->isShared();
```

Comment:
Look like this change break the compilation with `Undefined offset: 0` when the node doesn't have any `InEdges`

Reply:
hotfixed in https://github.com/symfony/symfony/commit/5eb17e55d2c3eb5736c35cc26d95189ac2d41f14
but I don't understand how this can happen logically (we're here because we did find a reference to the service, to it must have at least one in-edge.)
could you share a reproduder?

Reply:
The use case is: I use a custom TestKernel which inject mocks in the container and set the service as Synthetic with a compiler pass

NB: Changing the type of the compiler pass to `PassConfig::TYPE_AFTER_REMOVING` (to run after `InlineServiceDefinitionsPass`) fix the issue

This minimal reproduct case is:
```
class Kernel extends BaseKernel implements CompilerPassInterface
{
    // ...
    public function process(ContainerBuilder $container)
    {
        // routing.resolver is just an example for the minimal reproduct case here
        $container->getDefinition('routing.resolver')->setPublic(true)->setSynthetic(true);
    }
}
```

Reply:
ok, so this means we are missing a check for synthetic service!
added in https://github.com/symfony/symfony/pull/27366